internal/search/job/jobutil/feeling_lucky_search_job.go

```
@@ -398,6 +400,157 @@ func typePatterns(b query.Basic) *query.Basic {
+			pathValue = fmt.Sprintf("^%s$", pathValue)
+		} else if pathElems[2] == "tree" {
+			pathValue = fmt.Sprintf("^%s", pathValue)
+		} else {
+			// Don't know what this is.
+			return nil
+		}
+
+		return &[]query.Node{
+			query.Parameter{
+				Field:      query.FieldRepo,
+				Value:      repoValue,
+				Negated:    negated,
+				Annotation: query.Annotation{},
+			},
+			query.Parameter{
+				Field:      query.FieldRev,
+				Value:      revision,
+				Negated:    negated,
+				Annotation: query.Annotation{},
+			},
+			query.Parameter{
+				Field:      query.FieldFile,
+				Value:      pathValue,
+				Negated:    negated,
+				Annotation: query.Annotation{},
+			},
+		}
+	}
+
+	return nil
+}
+
+// patternsToCodeHostFilters converts patterns to `repo` or `path` filters if they
+// can be interpreted as URIs.
+func patternsToCodeHostFilters(b query.Basic) *query.Basic {
+	rawPatternTree, err := query.Parse(query.StringHuman([]query.Node{b.Pattern}), query.SearchTypeStandard)
+	if err != nil {
+		return nil
+	}
+
+	filterParams := []query.Node{}
+	changed := false
+	newParseTree := query.MapPattern(rawPatternTree, func(value string, negated bool, annotation query.Annotation) query.Node {
+		if params := patternToCodeHostFilters(value, negated); params != nil {
+			changed = true
+			filterParams = append(filterParams, *params...)
+			// Collect the param and delete pattern. We're going to
+			// add those parameters after. We can't map patterns
+			// in-place because that might create parameters in
+			// concat nodes.
+			return nil
+		}
+
+		return query.Pattern{
+			Value:      value,
+			Negated:    negated,
+			Annotation: annotation,
+		}
+	})
+
+	if !changed {
+		return nil
+	}
+
+	newParseTree = query.NewOperator(append(newParseTree, filterParams...), query.And) // Reduce with NewOperator to obtain valid partitioning.
+	newNodes, err := query.Sequence(query.For(query.SearchTypeStandard))(newParseTree)
+	if err != nil {
+		fmt.Printf("err 1")
```

```suggestion
```

---

Also, this _shouldn't_ ever error, right? I'm slightly concerned that just swallowing these errors will cover up the fact that our rules aren't doing anything if something breaks.

---

I was debugging something. In principle rules should be set up so that they shouldn't ever error, yes. so the concern is legitimate but in absence of logging and making this feature super duper duper robust (I really just want to roll i tout at this point) I'm OK punting here.

So I'll tell you how you can get an error here :-) I created a query with an `or` expression for a different rule and then when it tries to partition to a single basic query this fails. Then I went "d'oh, right I can't do that because generated queries can only be basic 😢 😢". I'm not going to do the work to change that right now, so just leaving out that idea.

---

Oh actually it wasn't this one that failed, it was the other one `newBasic, err := query.ToBasicQuery(newNodes)`

---

Cool, yep, I'm totally cool with punting